### PR TITLE
Update `release-v2.5` branch to pin to Mapbox Common `v21.3.0`.

### PR DESCRIPTION
**Pull request by MaximAlien**
### Description

Similar PR was created for `release-v2.4` branch in https://github.com/mapbox/mapbox-navigation-ios/pull/4048.

### CODE REVIEW

File: MapboxCoreNavigation.podspec [hunk 1]
```diff
@@ -43,6 +43,7 @@ Pod::Spec.new do |s|
   s.dependency "MapboxNavigationNative", ">= 101.0.1", "< 102.0.0"
   s.dependency "MapboxDirections", "~> 2.5.0"
```
Comment (Udumft) [inline MapboxCoreNavigation.podspec:44]: Should be bumped to 2.6.0?

File: Package.swift [hunk 2]
```diff
@@ -26,6 +26,7 @@ let package = Package(
         .package(name: "MapboxMobileEvents", url: "https://github.com/mapbox/mapbox-events-ios.git", from: "1.0.0"),
         .package(name: "MapboxNavigationNative", url: "https://github.com/mapbox/mapbox-navigation-native-ios.git", from: "101.0.1"),
         .package(name: "MapboxMaps", url: "https://github.com/mapbox/mapbox-maps-ios.git", from: "10.5.0"),
+        .package(name: "MapboxCommon", url: "https://github.com/mapbox/mapbox-common-ios.git", .exact("21.3.0")),
```
Comment (Udumft) [inline Package.swift:29]: Should we update `MapboxDirections` too?

**Comment by azarovalex** [comment]
Xcode 12.5.1 should still be available on CircleCI, can we keep the SPM test job just to verify everything builds on CI?

**Comment by MaximAlien** [comment]
> Xcode 12.5.1 should still be available on CircleCI, can we keep the SPM test job just to verify everything builds on CI?

It's still available by CircleCI, but no longer supported by Maps SDK, so we no longer fully support Xcode 12.x anyway.

File: MapboxCoreNavigation.podspec (hunk 1 above)
Comment (MaximAlien) [reply MapboxCoreNavigation.podspec:44]: I think that Directions `v2.6.0` will be automatically picked up anyway if we select up to next minor version. If you feel that it's useful to get new stuff from Directions `v2.6.0` even for this specific version I can update it.

**Review by jill-cardamon — APPROVED** [review]
lgtm unless you think we should explicitly update to MapboxDirections v2.6.0

**Pull request merged**